src/metabase/models/api_key.clj

```diff
@@ -75,17 +76,26 @@
     unhashed_key (assoc :key (u.password/hash-bcrypt unhashed_key))
     true (dissoc :unhashed_key)))
 
+(defn- add-updated-by-id [api-key]
+  (update api-key :updated_by_id #(or % api/*current-user-id*)))
+
+(defn- add-created-by-id [api-key]
+  (update api-key :created_by_id #(or % api/*current-user-id*)))
```

Comment:
Wouldn't this _not_ modify `:updated_by_id` once it's set for the first time? The anonymous fn takes the old value as an arg, so it'll always be used preferentially, right?

Reply:
Also — I think generally we're trying to move away from using the dynamic vars in `metabase.api.common` in non-API namespaces to make things more explicit and REPL-friendly. (Granted, we have a lot of it at the moment, but there's been some recent discussions about curbing it, because it's annoying to constantly have to bind `*current-user-id*` during development)

I won't block the PR over this, but I think it would be slightly better to keep these in the API namespace like they were before. Or add a functions in the model namespace to create/update API keys which take a user ID as an argument, similar to `metabase.models.card/create-card!`

Reply:
Good catch - and keeping `api` out of the models does seem more hygienic, I'll do that.